common/v2/features/ReceiveAssets/components/QRCode.tsx

```diff
@@ -0,0 +1,71 @@
+import QRCodeLib from 'qrcode';
+import React from 'react';
+
+// FIXME should store limited amount if history
+// data -> qr cache
+const cache: { [key: string]: string } = {};
+
+interface Props {
+  data: string;
+}
+
+interface State {
+  qr?: string;
+}
+
+export default class QRCode extends React.PureComponent<Props, State> {
```

Comment:
I suspect this is copied from the previous version.
Since this is useful for the rest of the app, please move to `v2/components`
Ideally we would refactor to remove the UNSAFE lifecycle hooks.

Reply:
It is copied from the previous app but I don't believe that we have any uses elsewhere for QRCodes at the moment. I think that this should be moved to a shared directory if we know it'll be used elsewhere.

Reply:
Since it has the potential to be shared, I would prefer it under `components`.
Especially because the developer that would code the new feature may not be aware of 'ReceiveAssets' when he starts so his reflex would be to code a new one.
I think that by encouraging reusable components it creates better code (pure components with a better SOC) and the dev will be less inclined to take shortcuts and mingle everything together.

Reply:
Oh that makes sense! I'll do that then!

Reply:
We actually use QRCode from old `components/ui` in v2 PaperWallet (https://github.com/MyCryptoHQ/MyCrypto/blob/gau/common/v2/components/PaperWallet.tsx#L7). Please update imports and check if everything still works (it should since it is an exact copy).